lang/csharp/src/apache/main/CodeGen/CodeGen.cs

```diff
@@ -219,8 +219,14 @@ protected virtual void processProtocols()
         /// <returns>
         /// List of named schemas.
         /// </returns>
-        protected virtual SchemaNames generateNames(Protocol protocol)
+        /// <exception cref="System.ArgumentNullException">protocol - Protocol can not be null.</exception>
+        protected virtual SchemaNames GenerateNames(Protocol protocol)
         {
+            if (protocol == null)
+            {
+                throw new ArgumentNullException(nameof(protocol), "Protocol can not be null");
```

Comment:
What does `nameof()` returns for `null` ?

Reply:
Honestly, it's a weird pattern, but really it's to make sure your messaging is up to date when you change the name of a variable.  It's the equivalent of "protocol."  It's literally the name of the variable as a string.